src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -76,6 +77,27 @@ void setUserStore(UserStore userStore) {
     this.userStore = userStore;
   }
 
+  /**
+   * This function formats any URL's within a message into HTML links.
+   */
+  String formatURL(String messageContent){
+    // URL regex
+    String patternString = "(http://www\\.|https://www\\.|http://|https://)+[a-z0-9]+([\\-\\.]{1}[a-z0-9]+)*\\.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?";
+    Pattern pattern = Pattern.compile(patternString);
+    Matcher m = pattern.matcher(messageContent);
+    // Used to rebuild messageContent
+    StringBuffer sb = new StringBuffer(messageContent.length());
+
+    // Convert each url found into an HTML link
+    while(m.find()){
+      String url = m.group();
+      m.appendReplacement(sb, "<a href = " + url + ">" + url + "</a>");
+    } 
+    m.appendTail(sb);
+
+    return sb.toString();
+  }
+
```

Comment:
This might be a little nitpicky, but try not to use names like 'm' and 'sb' in the code. Change it to something that readers of the code will be able to immediately grasp what it does without having to travel back to the source to find the type